components/InputField.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { Text, TextInput, StyleSheet, TextInputProps, ViewStyle } from 'react-native';
+import { Control, Controller, FieldError } from 'react-hook-form';
+import Colors from '../constants/Colors';
+
+
+type InputFieldProps = {
+  error?: FieldError;
+  control: Control<any, any>;
+  name: string;
+  defaultValue?: string;
+  placeholder?: string;
+  secureTextEntry?: boolean;
+  multiline?: boolean;
+  bigTextBox?: boolean;
+  inputTitle?: string;
+  autoCapitalize?: TextInputProps['autoCapitalize'];
+  keyboardType?: TextInputProps['keyboardType'];
+};
+
+export default function InputField({
+  error,
+  control,
+  name,
+  defaultValue = '',
+  placeholder = '',
+  secureTextEntry = false,
+  multiline = false,
+  bigTextBox = false,
+  inputTitle = undefined,
+  autoCapitalize = undefined,
+  keyboardType = undefined,
+}: InputFieldProps) {
+
+  let textInputStyle: ViewStyle[] = [styles.inputBox];
+  if (bigTextBox) textInputStyle.push(styles.h120);
+  if (inputTitle == undefined) textInputStyle.push(styles.mt16);
```

Comment:
I'm not 100% why you need to do these conditional styles but we could do this same logic like this:

`const style = [styles.inputBox, bigTextBox && styles.h120, inputTitle == undefined && styles.mt16].filter(Boolean)`